src/views/MissionPlanningView.vue

```diff
@@ -694,102 +1195,241 @@ watch(isCreatingSurvey, (isCreatingNow) => {
 })
 
 const generateWaypointsFromSurvey = (): void => {
-  if (!surveyPathLayer.value) {
-    showDialog({ variant: 'error', message: 'No survey path to generate waypoints from.', timer: 3000 })
+  if (!surveyPolygonVertexesPositions.value.length) {
+    console.log('No survey polygon defined.')
     return
   }
 
-  const surveyLatLngs = surveyPathLayer.value.getLatLngs()
-  if (!Array.isArray(surveyLatLngs) || surveyLatLngs.length === 0) {
-    showDialog({ variant: 'error', message: 'Invalid survey path.', timer: 3000 })
+  const newSurveyId = uuid()
+
+  canUndo.value[newSurveyId] = true
+
+  const polygonCoordinates: WaypointCoordinates[] = surveyPolygonVertexesPositions.value.map((latLng) => [
+    latLng.lat,
+    latLng.lng,
+  ])
+
+  lastSurveyState.value[newSurveyId] = {
+    polygonPositions: polygonCoordinates,
+  }
+
+  const adjustedAngle = 90 - surveyLinesAngle.value
+  const continuousPath = generateSurveyPath(
+    surveyPolygonVertexesPositions.value,
+    distanceBetweenSurveyLines.value,
+    adjustedAngle
+  )
+
+  if (!continuousPath.length) {
+    showDialog({
+      variant: 'error',
+      message: 'No valid path could be generated. Try adjusting the angle or distance between lines.',
+      timer: 3000,
+    })
     return
   }
 
-  surveyWaypoints.value = []
```

Comment:
It's important to mention: there's a big block of code here being interpreted as if had been removed and replaced with something else completely different, and this is happening because of the change from the `.flat().forEach()` to the `.map()` strategy.

Both strategies lead to the same result and there was no need to change it. This kind of change ends up marking hundreds of lines more as diffs and increasing the review workload by an infinitude, specially on PRs like this.

It would help not doing those kind of changes in the future, to prevent the unnecessary workload.